tracing-appender/src/lib.rs

```diff
@@ -1,48 +1,127 @@
+//! [tracing]: https://docs.rs/tracing/
+//!
+//! # Usage
+//!
+//! Add the following to your `Cargo.toml`:
+//! ```toml
+//! tracing-appender = "0.1"
+//! ```
+//!
+//! This crate can be used in a few ways to record spans/events:
+//!  - Using a [`RollingFileAppender`][rolling_struct] to perform writes to a log file. This will block on writes.
+//!  - Using *any* type implementing [`std::io::Write`][write] in a non-blocking fashion.
+//!  - Using a combination of [`NonBlocking`][non_blocking] and [`RollingFileAppender`][rolling_struct] to allow writes to a log file
+//! without blocking.
+//!
+//! ## Rolling File Appender
+//!
+//! ```rust
+//! # fn docs() {
+//! let file_appender = tracing_appender::rolling::hourly("/some/directory", "prefix.log");
+//! # }
+//! ```
+//! This creates an hourly rotating file appender that writes to `/some/directory/prefix.log.YYYY-MM-DD-HH`.
+//! [`Rotation::DAILY`] and [`Rotation::NEVER`] are the other available options.
+//!
+//! The file appender implements [`std::io::Write`][write]. To be used with [`tracing_subscriber::FmtSubscriber`][fmt_subscrier],
+//! it must be combined with a [`MakeWriter`][make_writer] implementation to be able to record tracing spans/event.
+//!
+//! The [`rolling` module][rolling]'s documentation provides more detail on how to use this file appender.
+//!
+//! ## Non-Blocking Writer
+//!
+//! The example below demonstrates the construction of a `non_blocking` writer with `std::io::stdout()`,
+//! which implements [`MakeWriter`][make_writer].
+//!
+//! ```rust
+//! # fn doc() {
+//! let (non_blocking, _guard) = tracing_appender::non_blocking(std::io::stdout());
+//! let subscriber = tracing_subscriber::fmt().with_writer(non_blocking);
+//! tracing::subscriber::set_global_default(subscriber.finish()).expect("Could not set global default");
+//! # }
+//! ```
+//! **Note:** `_guard` is a [`WorkerGuard`][guard] which is returned by [`tracing_appender::non_blocking`][non_blocking]
+//! to ensure buffered logs are flushed to their output in the case of abrupt terminations of a process.
+//! See [`WorkerGuard` module][guard] for more details.
+//!
+//! The example below demonstrates the construction of a [`tracing_appender::non_blocking`][non_blocking]
+//! writer constructed with a [`std::io::Write`][write]:
+//!
+//! ```rust
+//! use std::io::Error;
+//!
+//! struct TestWriter;
+//!
+//! impl std::io::Write for TestWriter {
+//!     fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+//!         let buf_len = buf.len();
+//!         println!("{:?}", buf);
+//!         Ok(buf_len)
+//!     }
+//!
+//!     fn flush(&mut self) -> std::io::Result<()> {
+//!         Ok(())
+//!     }
+//! }
+//!
+//! # fn doc() {
+//! let (non_blocking, _guard) = tracing_appender::non_blocking(TestWriter);
+//! let subscriber = tracing_subscriber::fmt().with_writer(non_blocking);
+//! tracing::subscriber::set_global_default(subscriber.finish()).expect("Could not set global default");
```

Comment:
and, again, maybe consider:
```suggestion
//! tracing_subscriber::fmt()
//!     .with_writer(non_blocking)
//!     .init();
```